src/ServiceControl.Transports.AzureServiceBus/QueueLengthProvider.cs

```diff
@@ -0,0 +1,32 @@
+namespace ServiceControl.Transports.AzureServiceBus
+{
+    using System.Threading.Tasks;
+    using Monitoring;
+    using Monitoring.Infrastructure;
+    using Monitoring.Messaging;
+    using Monitoring.QueueLength;
+    using NServiceBus.Metrics;
+
+    public class QueueLengthProvider : IProvideQueueLength
+    {
+        public void Initialize(string connectionString, QueueLengthStore store)
```

Comment:
Is idea to have provider per namespace?

Reply:
The current set up would look something like this:

- Create a new Monitoring Instance. Configure it with the Azure Service Bus transport and a connection string
- Install the NServiceBus.Metrics.ServiceControl plugin into your endpoints and configure it to send metric data to the Monitoring Instance created in the previous step

We don't specify namespace anywhere. Is that a part of the connection string? Or a part of the logical address? Or something else entirely?

For the other metrics, this isn't a problem. I suspect that all communication between the endpoint and the monitoring instance use a single namespace BUT if an endpoint is bound to queues in multiple namespaces then we need some way of telling the Monitoring Instance about that.

Reply:
> We don't specify namespace anywhere. Is that a part of the connection string? 

Connection string. Multiple connection strings can only be declared via code.

Reply:
> if an endpoint is bound to queues in multiple namespaces then we need some way of telling the Monitoring Instance about that.

And that was my original question asked [8 days ago](https://github.com/Particular/ServiceControl.Monitoring/pull/124#pullrequestreview-118983011) 😄 Conway's law FTW 😂